lib/crawler/api/config.rb

```diff
@@ -277,25 +275,34 @@ def load_certificate_from_file(file_name)
 
       #---------------------------------------------------------------------------------------------
       def configure_domain_allowlist!
-        raise ArgumentError, 'Needs at least one domain' unless domain_allowlist&.any?
+        raise ArgumentError, 'Needs at least one domain' unless domains&.any?
+
+        @domain_allowlist = domains.map do |domain|
```

Comment:
why maintain `@domain_allowlist` and `@seed_urls` instead of having a `@domains` instance variable that matches the configuration?

Reply:
I was concerned about how much code would need to be updated to do that. Not that that's bad in itself, but it does make PR reviewing daunting.
I'll look at how impactful changing this is and get back to you before I merge this.

Reply:
Actually I remember now; because `domains` can have multiple entries, we need to collect all of the values for `domains[].url` and `domains[].seed_urls` so the crawler can do them concurrently. It's easier to just have them as a pooled value.